canvas_cli/apps/plugin/plugin.py

```diff
@@ -503,6 +508,81 @@ def set_secrets(
     update(name=plugin, package_path=None, secrets=secrets, host=host, is_enabled=None)
 
 
+def _find_unreferenced_handlers(plugin_path: Path, manifest_json: dict) -> builtins.list[str]:
+    """Find handler classes that aren't referenced in the manifest."""
+    # Get all handler/protocol class references from manifest
+    components = manifest_json.get("components", {})
+    referenced_classes = set()
+
+    # Check both "handlers" and "protocols" keys for backwards compatibility
+    for key in ["handlers", "protocols"]:
+        for item in components.get(key, []):
+            class_ref = item.get("class", "")
+            # Store just the class part (module.path:ClassName)
+            referenced_classes.add(class_ref)
+
+    # Find all Python files in the plugin
+    unreferenced = []
+    python_files = builtins.list(plugin_path.rglob("*.py"))
+
+    # Import BaseHandler to check subclasses
+    try:
+        from canvas_sdk.handlers import BaseHandler
+    except ImportError:
+        # If we can't import BaseHandler, skip the check
+        return []
+
+    for py_file in python_files:
+        # Skip test files, __pycache__, and __init__ files
+        # Check if file is in a test directory or is a test file itself
+        parts = py_file.parts
+        if "__pycache__" in str(py_file) or py_file.name == "__init__.py":
+            continue
+        if "tests" in parts or "test" in parts or py_file.stem.startswith("test_"):
+            continue
+
+        # Try to import and inspect the module
+        try:
+            # Create a unique module name
+            module_name = f"temp_module_{py_file.stem}"
+            spec = importlib.util.spec_from_file_location(module_name, py_file)
+            if spec and spec.loader:
+                module = importlib.util.module_from_spec(spec)
```

Comment:
I think this is dangerous. 

Importing everything on a plugin without sandboxing could _probably_ be used as an attack vector for bad actors. 

@beaugunderson you know way more about security than I do, would love your thoughts.